src/client/datascience/notebook/intellisense/pythonKernelCompletionProvider.ts

```diff
@@ -91,13 +91,14 @@ export class PythonKernelCompletionProvider implements CompletionItemProvider {
         // Allow slower timeouts for CI (testing).
         const timeout =
             parseInt(process.env.VSC_JUPYTER_IntellisenseTimeout || '0', 10) || Settings.IntellisenseTimeout;
+        const pylanceTimeout = Math.min(1000, timeout);
         traceInfoIfCI(`Notebook completion request for ${document.getText()}, ${document.offsetAt(position)}`);
         const [result, pylanceResults] = await Promise.all([
             waitForPromise(
                 this.getJupyterCompletion(kernel.session, document.getText(), document.offsetAt(position), token),
                 timeout
             ),
-            this.getPylanceCompletions(document, position, context, token)
+            waitForPromise(this.getPylanceCompletions(document, position, context, token), pylanceTimeout)
```

Comment:
Could someone have a situation where it consistently takes just around that much time for pylance to return results? Would the result be that sometimes the completion list will contain pylance results and sometimes it doesn't?